app.py

```diff
@@ -2,14 +2,36 @@
 Flask entry point
 """
 
-from flask import Flask
+import os
+from flask import Flask, Blueprint
+from flask_restful import Api, Resource
+import config
+from db.connectionManager import ConnectionManager
+from api.v1.listing import Listing
 
+CONFIGS = {
+    'production': config.ProdConfig,
+    'dev': config.DevConfig,
+    'test': config.TestConfig,
+    'default': config.DevConfig
+}
+config_name = os.getenv('FLASK_CONFIGURATION', 'default')
+
+# Setup Flask Environment
 app = Flask(__name__)
+app.config.from_object(CONFIGS[config_name])
+api_blueprint = Blueprint('api', __name__)
+api = Api(api_blueprint)
+print(app.config['STARTUP_MSG'])
 
-@app.route('/listing', methods=['POST'])
-def listing():
-    return 'OK'
+# Initialize DB Connection
+db = ConnectionManager(app.config['DB_URL'], app.config['LOCAL'], app.config['TABLE_NAME'], app.config['REGION'])
+
+# Endpoints
+api.add_resource(Listing, '/api/v1/listing', resource_class_kwargs={'db': db})
```

Comment:
interesting... is this like ye olde rails resourceful routes? ah, ic, its wrapped in the class above. so i'm guessing you'd have whichever CRUD methods on that class you wanted to expose

if you don't write a method for a verb on that class and it gets called what happens? 404?

Reply:
Yup. 404